pkg/types/types.go

```diff
@@ -40,6 +40,7 @@ type NetConf struct {
 	SocketFile             string   `json:"socket_file"`
 	LinkStateCheckRetries  int      `json:"link_state_check_retries"`
 	LinkStateCheckInterval int      `json:"link_state_check_interval"`
+	DPDKMode               bool     `json:"-,omitempty"`
```

Comment:
It is not clear what you trying to achieve with \`json:"-,omitempty"\` syntax. If you want to hide the field from the API, then the correct syntax is just \`json:"-"\`. Currently DPDKMode renders as "-" in the cached json.

I see two options here:
1) use meaningful json filed name for DPDKMode in NetConf struct 
2) move DPDKMode field to the CachedNetConf struct

The first option will pollute the user API. My personal preference is to use opt 2.

Reply:
Done. I also prefer the second option.
Additionally, I changed `DPDKMode` to `UserspaceMode`, as the VF can be used not only for containerized DPDK but also for containerized QEMU, such as in KubeVirt.

https://github.com/k8snetworkplumbingwg/ovs-cni/blob/074e96538c17b1e5f5902b58e7c65a06f73f62a6/pkg/types/types.go#L74-L83